pkg/compact/compact.go

```diff
@@ -194,6 +198,9 @@ func (c *Syncer) syncMetas(ctx context.Context) error {
 					continue
 				}
 				if err != nil {
+					if removedOrIgnored := c.removeIfMetaMalformed(workCtx, id); removedOrIgnored {
```

Comment:
Well technically you can do just `if c.removeIfMetaMalformed(workCtx, id)`  here (: